### ISSUE TITLE
Jetty 12 does not deploy my war, but Jetty 11 and Tomcat 11.0.0-jre2 (run in docker container)

### ISSUE THREAD

**Issue by halacs** [eclipse/jetty.project#11652]
Jetty 12 does not deploy my root.war. If I switch back to Jetty:11 or use tomcat:11 same root.war file works as expected. I use jakarta.servlet-api-6.1.0-M2 and GWT-2.11.0

```
root@gabor-VirtualBox:/tmp/webapps# ll
total 76128
drwxrwxrwx  2 root  root      4096 ápr   14 16:50 ./
drwxrwxrwt 20 root  root      4096 ápr   14 16:48 ../
-rw-rw-r--  1 gabor gabor 77943566 ápr   14 16:23 root.war
```

```
root@gabor-VirtualBox:/tmp/webapps# docker run --net host --rm --name jetty -it -v /tmp/jetty -v /run/jetty -v /tmp/webapps:/var/lib/jetty/webapps jetty:12 --add-modules=http,ee10-deploy,ee10-webapp
/opt/java/openjdk/bin/java -Djava.io.tmpdir=/tmp/jetty -Djetty.home=/usr/local/jetty -Djetty.base=/var/lib/jetty -Djava.io.tmpdir=/tmp/jetty --class-path /var/lib/jetty/resources:/usr/local/jetty/lib/logging/slf4j-api-2.0.12.jar:/usr/local/jetty/lib/logging/jetty-slf4j-impl-12.0.8.jar:/usr/local/jetty/lib/jetty-http-12.0.8.jar:/usr/local/jetty/lib/jetty-server-12.0.8.jar:/usr/local/jetty/lib/jetty-xml-12.0.8.jar:/usr/local/jetty/lib/jetty-util-12.0.8.jar:/usr/local/jetty/lib/jetty-io-12.0.8.jar org.eclipse.jetty.xml.XmlConfiguration java.io.tmpdir=/tmp/jetty java.version=21.0.2 jetty.base=/var/lib/jetty jetty.base.uri=file:///var/lib/jetty jetty.home=/usr/local/jetty jetty.home.uri=file:///usr/local/jetty jetty.webapp.addServerClasses=org.eclipse.jetty.logging.,file:///usr/local/jetty/lib/logging/,org.slf4j. runtime.feature.alpn=true slf4j.version=2.0.12 /usr/local/jetty/etc/jetty-bytebufferpool.xml /usr/local/jetty/etc/jetty-threadpool.xml /usr/local/jetty/etc/jetty.xml /usr/local/jetty/etc/jetty-http.xml
INFO  : http already enabled by [${jetty.base}/start.d/http.ini]
INFO  : sessions        transitively enabled, ini template available with --add-modules=sessions
INFO  : security        transitively enabled
INFO  : ee10-deploy     initialized in ${jetty.base}/start.d/ee10-deploy.ini
INFO  : ee10-security   transitively enabled
INFO  : ee10-webapp     initialized in ${jetty.base}/start.d/ee10-webapp.ini
INFO  : ee10-servlet    transitively enabled
INFO  : deploy          transitively enabled
INFO  : Base directory was modified
2024-04-14 14:50:49.863:INFO :oejs.Server:main: jetty-12.0.8; built: 2024-03-29T19:58:19.443Z; git: ffffdcc3c863b8f49901cc7c84a8157ebc5d15f9; jvm 21.0.2+13-LTS
2024-04-14 14:50:49.889:INFO :oejs.AbstractConnector:main: Started ServerConnector@3b07a0d6{HTTP/1.1, (http/1.1)}{0.0.0.0:8080}
2024-04-14 14:50:49.899:INFO :oejs.Server:main: Started oejs.Server@6328d34a{STARTING}[12.0.8,sto=5000] @336ms
^C2024-04-14 14:51:08.142:INFO :oejs.Server:JettyShutdownThread: Stopped oejs.Server@6328d34a{STOPPING}[12.0.8,sto=5000]
2024-04-14 14:51:08.142:INFO :oejs.Server:JettyShutdownThread: Shutdown oejs.Server@6328d34a{STOPPING}[12.0.8,sto=5000]
2024-04-14 14:51:08.145:INFO :oejs.AbstractConnector:JettyShutdownThread: Stopped ServerConnector@3b07a0d6{HTTP/1.1, (http/1.1)}{0.0.0.0:8080}
```

**Comment by sbordet** [comment]
Are you sure the war is in the proper directory?
Seems in `/tmp/webapps`, but your JETTY_BASE is `/var/lib/jetty`?
The logs seem to show that nothing was deployed because there was nothing to deploy.

**Comment by halacs** [comment]
Yes, 99% :)

I run in Docker. `/tmp/webapps` directorry of the host is mounted into `/var/lib/jetty/webapps` directory of the container.

If I change nothing else only the container image tag from `jetty:12` to `jetty:11` all works fine.

**Comment by sbordet** [comment]
Can you enable DEBUG logging for Jetty and report the logs?

**Comment by halacs** [comment]
Can you please help me where to enable it?

**Comment by sbordet** [comment]
https://eclipse.dev/jetty/documentation/jetty-12/operations-guide/index.html#og-troubleshooting-logging

**Comment by halacs** [comment]
Here are the logs generated by the first command below:
[jetty_202404142009.log](https://github.com/jetty/jetty.project/files/14971643/jetty_202404142009.log)

```
root@gabor-VirtualBox:~# docker run --net host --rm --name jetty -it -v /tmp/jetty -v /run/jetty -v /tmp/webapps:/var/lib/jetty/webapps -v /root/jetty/jetty-logging.properties:/var/lib/jetty/resources/jetty-logging.properties jetty:12 &> /tmp/jetty_202404142009.log
```

To make sure, root.war is at the right place inside the container:

```
root@gabor-VirtualBox:~/jetty# docker exec -it jetty ls -lah /var/lib/jetty/webapps
total 75M
drwxrwxrwx 2 root  root  4.0K Apr 14 14:50 .
drwxr-xr-x 1 jetty jetty 4.0K Apr 14 18:09 ..
-rw-rw-r-- 1  1000  1000  75M Apr 14 15:17 root.war
```

I increased the log level by `org.eclipse.jetty.LEVEL=DEBUG` in the below file then mounting it into the above container with the `-v` argument.

```
root@gabor-VirtualBox:~/jetty# cat jetty-logging.properties 
## Set logging levels from: ALL, TRACE, DEBUG, INFO, WARN, ERROR, OFF
#org.eclipse.jetty.LEVEL=INFO
org.eclipse.jetty.LEVEL=DEBUG
## Configure a level for an arbitrary logger tree
#com.example.LEVEL=INFO
## Configure a level for specific logger
#com.example.MyComponent.LEVEL=INFO
## Configure JMX Context Name
# org.eclipse.jetty.logging.jmx.context=JettyServer
## Hide stacks traces in an arbitrary logger tree
#com.example.STACKS=false
```

**Comment by sbordet** [comment]
There is no trace of the `ee10-deployer` in the logs.
Are you sure you configured the JETTY_BASE properly?

Please run `java -jar start.jar --list-config` from within your JETTY_BASE.

I'm quite confident it's just a simple configuration mistake.

**Comment by halacs** [comment]
No, I'm not sure at all but tried to do so.

[jetty_202404142029.log](https://github.com/jetty/jetty.project/files/14971718/jetty_202404142029.log)

```
root@gabor-VirtualBox:~# docker run --net host --rm --name jetty -it -v /tmp/jetty -v /run/jetty -v /tmp/webapps:/var/lib/jetty/webapps -v /root/jetty/jetty-logging.properties:/var/lib/jetty/resources/jetty-logging.properties jetty:12 --list-config &> /tmp/jetty_202404142029.log
```

**Comment by sbordet** [comment]
@halacs see by yourself, you don't have the `ee10-deploy` module enabled in your JETTY_BASE.

**Comment by halacs** [comment]
`--add-modules=ee10-deploy` should enable it, right?

**Comment by halacs** [comment]
Thanks, I found the problem.

`--add-modules=ee10-deploy` created a sample config file under the name of `/var/lib/jetty/start.d/ee10-deploy.ini` but it was not enough.

Usually the example configs are the default settings too when not declared the opposite. This is not true here. Here I had to remove the comment sign from the beginning of a few lines and now my .war is loaded properly. Thanks for pointing our the missing modul.
